### PR TITLE
Add Xcode template for features with @ObservableState

### PR DESCRIPTION
**Pull request by matiasbzurovski**
Adding it as a new template because we still may need to use the _old_ version when creating a feature that uses other one which doesn't have @Reducer and @ObservableState.

### CODE REVIEW

File: templates/Xcode/TCA Observable Feature.xctemplate/___FILEBASENAME___.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	}
+
+	public typealias Action = FeatureAction<Self>
+
+	public enum ViewAction: Sendable, Equatable {
+		case appeared
+	}
+
+	public init() {}
+
```
Comment (CyonAlexRDX) [inline templates/Xcode/TCA Observable Feature.xctemplate/___FILEBASENAME___.swift:16]: Should we not include the body here explicitly? I think @danvleju-rdx found out it was needed? Or I might misremember

**Review by CyonAlexRDX — APPROVED** [review]
Nice

File: templates/Xcode/TCA Observable Feature.xctemplate/___FILEBASENAME___.swift (hunk 1 above)
Comment (matiasbzurovski) [reply templates/Xcode/TCA Observable Feature.xctemplate/___FILEBASENAME___.swift:16]: I've seen it as well, but I am able to make it compile without it. @danvleju-rdx, if it compiles without it does it mean we can drop it, or was it added for runtime behavior differences?

Comment (matiasbzurovski) [reply templates/Xcode/TCA Observable Feature.xctemplate/___FILEBASENAME___.swift:18]: just realised it was the second, so added it to template

Comment (danvleju-rdx) [reply templates/Xcode/TCA Observable Feature.xctemplate/___FILEBASENAME___.swift:18]: Yes, runtime issues, [see](https://github.com/radixdlt/babylon-wallet-ios/pull/1315)

**Review by danvleju-rdx — APPROVED** [review]

**Pull request merged**